crates/db_schema/src/lib.rs

```diff
@@ -119,3 +121,17 @@ impl From<Url> for DbUrl {
 pub fn naive_now() -> NaiveDateTime {
   chrono::prelude::Utc::now().naive_utc()
 }
+
+#[repr(transparent)]
+#[derive(Clone, PartialEq, Serialize, Deserialize, Debug, AsExpression, Queryable)]
+#[sql_type = "Text"]
+pub struct PrimaryLanguageTag(String);
```

Comment:
@dessalines Do you know how I can get this to compile? Nothing I tried worked, and the errors are quite confusing.

https://cloud.drone.io/LemmyNet/lemmy/874/1/5

Reply:
I really dislike trying to force these other types into diesel... its never guaranteed to work. I'd just leave them as strings, and do the error checking and parsing at the API level.